docs/resources/repo.md

```diff
@@ -30,6 +30,15 @@ The following arguments are supported:
 * `branch` - (Optional) name of the branch for initial checkout. If not specified, the default branch of the repository will be used.  Conflicts with `tag`.  If `branch` is removed, and `tag` isn't specified, then the repository will stay at the previously checked out state.
 * `tag` - (Optional) name of the tag for initial checkout.  Conflicts with `branch`.
 
+## sparse_checkout
+
+Optional `sparse_checkout` configuration block contains attributes related to [sparse checkout feature](https://docs.databricks.com/repos/git-operations-with-repos.html#configure-sparse-checkout-mode) in Databricks Repos.  It supports following attributes:
+
+* `patterns` - array of paths (directories) that will be used for sparse checkout.  List of patterns could be updated in-place.
+
+Addition or removal of the `sparse_checkout` configuration block will lead to recreation of the repo.
```

Comment:
Our API gives error when you're trying to add `sparse_checkout` to the existing repository - I double checked this. I'm happy to change it if you confirm that it shouldn't get an error.

Reply:
also, if you do first with `sparse_checkout` and then remove that block, it won't restore the full checkout

Reply:
The current API behavior is: 
- You cannot enable/disable sparse checkout after a repository is created, so if you create a repository without sparse checkout, setting sparse_checkout in Update request will fail.
-  If you specify sparse_checkout in Update request, it'll update the patterns to the new value; otherwise the sparse checkout setting won't be changed.

Thanks for pointing it out, I'll update the API doc to make it clearer.

Reply:
then yes, Terraform behaviour is matching these rules you can update list of the patterns for repo with `sparse_checkout`, but you can add or remove this block to the existing repository - it will be recreated with the new settings.